src/arkit.cs

```diff
@@ -856,20 +976,37 @@ interface ARFaceGeometry : NSCopying, NSSecureCoding {
 	[BaseType (typeof(SCNGeometry))]
 	[DisableDefaultCtor]
 	interface ARSCNFaceGeometry {
+		[Obsolete ("Use the 'Create' static constructor instead.")]
 		[Static]
 		[Export ("faceGeometryWithDevice:")]
 		[return: NullAllowed]
 		ARSCNFaceGeometry CreateFaceGeometry (IMTLDevice device);
 
+		[Static]
+		[Wrap ("CreateFaceGeometry (device)")]
+		ARSCNFaceGeometry Create (IMTLDevice device);
+
+		[Obsolete ("Use the 'Create' static constructor instead.")]
 		[Static]
 		[Export ("faceGeometryWithDevice:fillMesh:")]
 		[return: NullAllowed]
 		ARSCNFaceGeometry CreateFaceGeometry (IMTLDevice device, bool fillMesh);
 
+		[Static]
+		[Wrap ("CreateFaceGeometry (device, fillMesh)")]
+		ARSCNFaceGeometry Create (IMTLDevice device, bool fillMesh);
+
 		[Export ("updateFromFaceGeometry:")]
 		void Update (ARFaceGeometry faceGeometry);
 	}
 
+	[NoWatch, NoTV, NoMac, iOS (11,3)]
+	[BaseType (typeof(ARAnchor))]
+	interface ARImageAnchor {
```

Comment:
the base type `ARAnchor` has a `[DisableDefaultCtor]` and I don't see any reason for this subclass not to have it ?

Reply:
Mmh so `ARAnchor.h` has:

```
/**
 Initializes a new anchor object.
 @param transform The transformation matrix that defines the anchor’s rotation, translation and scale in world coordinates.
 */
- (instancetype)initWithTransform:(matrix_float4x4)transform;

/** Unavailable */
- (instancetype)init NS_UNAVAILABLE;
+ (instancetype)new NS_UNAVAILABLE;
```

`ARImageAnchor` and `ARFaceAnchor` (the only 2 subclasses) only have:

```
/** Unavailable */
- (instancetype)initWithTransform:(matrix_float4x4)transform NS_UNAVAILABLE;
```

Nothing is specified for `init` and `new` but given that `initWithTransform` is unavailable the must still have 1 constructor left (:

Reply:
yes, `init*` are methods in objc so they are inherited
and that's why we need to inline some `init*` in subclasses
still it means no `init` - it's the `initWithTransform1 that must be duplicated

Reply:
Oh yea ok so indeed `init` and `new` are also **unavailable** in the subclasses because they are inherited so `[DisableDefaultCtor]` makes sense (btw that could be caught by a test no? Unless the subclass in ObjC sometimes re-expose `init` and `new` even it's marked as unavailable in the parent?).

Now I don't follow your conclusion because `initWithTransform` is explicitly marked as unavailable. Looks like this type is not meant to be created by the user.